tests/test_distributed.py

```diff
@@ -25,3 +26,246 @@ def test_main(self):
         assert "x-datadog-sampling-priority" not in data["request_headers"]
         assert "x-datadog-tags" not in data["request_headers"]
         assert "x-datadog-trace-id" not in data["request_headers"]
+
+
+@scenarios.trace_propagation_style_w3c_datadog_b3
+@features.w3c_headers_injection_and_extraction
+class Test_Span_Links_From_Conflicting_Contexts:
+    """Verify headers containing conflicting trace context information are added as span links"""
+
+    def setup_span_links_from_conflicting_contexts(self):
+        extract_headers = {
+            "traceparent": "00-11111111111111110000000000000002-000000003ade68b1-01",
+            "tracestate": "dd=s:2;p:000000000000000a,foo=1",
+            "x-datadog-parent-id": "10",
+            "x-datadog-trace-id": "2",
+            "x-datadog-tags": "_dd.p.tid=2222222222222222",
+            "x-datadog-sampling-priority": "2",
+            "x-b3-traceid": "11111111111111110000000000000003",
+            "x-b3-spanid": "a2fb4a1d1a96d312",
+            "x-b3-sampled": "0",
+        }
+
+        self.req = weblog.get("/make_distant_call", params={"url": "http://weblog:7777"}, headers=extract_headers)
+
+    def test_span_links_from_conflicting_contexts(self):
+        trace = [
+            span
+            for _, _, span in interfaces.library.get_spans(self.req, full_trace=True)
+            if retrieve_span_links(span) is not None
+            and span["trace_id"] == 2  # Only fetch the trace that is related to the header extractions
```

Comment:
```suggestion
            and span["trace_id"] == 2  and span["parent_id"] == 10  # Only fetch the trace that is related to the header extractions
```

Reply:
This concept would work but value would need to be the tracecontext parent-id. I will add it in my next commit!